### PR TITLE
NSFS Fixes | handling too many forks exits + cleaning unwrap error in log

### PR DESCRIPTION
**Pull request by jackyalbo**
### Explain the changes
1. adding special handling in case of getting too many forks exits in a given time frame
2. adding read_bucket_sdk_config_info to remove errors when running nsfs in simple mode

### Issues: Fixed #xxx / Gap #xxx
1. 

### Testing Instructions:
1. 

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

File: src/util/fork_utils.js [hunk 1]
```diff
@@ -38,11 +38,22 @@ function start_workers(metrics_port, count = 0) {
...
+            console.warn('WORKER exit', { id: worker.id, pid: worker.process.pid, code, signal });
             new NoobaaEvent(NoobaaEvent.FORK_EXIT).create_event(undefined, { id: worker.id, pid: worker.process.pid,
                 code: code, signal: signal}, undefined);
+            // SIGABRT is used for abnormal program termination - https://www.qnx.com/developers/docs/6.4.1/dinkum_en/ecpp/stdlib.html
+            // we saw it happen both in case of no memory and on too much processes started
+            // in these case we won't restart the workers and will kill the main process
+            // for testing (linux only) limit memory(virt): ulimit -v xxx(kb) 
+            // limit user processes: ulimit -u xxx
+            if (signal === 'SIGABRT') {
+                console.error('EXIT ON WORKER ERROR');
```
Comment (romayalon) [inline src/util/fork_utils.js:50]: I would add a bit more info about the error
```suggestion
                console.error('EXIT ON WORKER ERROR received abort SIGABRT signal');
```

File: src/util/fork_utils.js [hunk 2]
```diff
@@ -38,11 +38,22 @@ function start_workers(metrics_port, count = 0) {
...
+                new NoobaaEvent(NoobaaEvent.ENDPOINT_CRASHED).create_event(undefined, { id: worker.id, pid: worker.process.pid,
```
Comment (romayalon) [inline src/util/fork_utils.js:51]: Without any message that says that we are restarting the main process, someone might think that this process crash happened unexpectedly. 
Maybe add to the arguments a message that describes that this is a crash by us or a new RESTART event, what ever works for you?

File: src/util/fork_utils.js [hunk 3]
```diff
@@ -38,11 +38,22 @@ function start_workers(metrics_port, count = 0) {
         // We don't want to leave the process with a partial set of workers,
         // so if any worker exits, we will print an error message in the logs and start a new one.
         cluster.on('exit', (worker, code, signal) => {
-            console.warn('WORKER exit', { id: worker.id, pid: worker.process.pid, code, signal }, 'starting a new one.');
+            console.warn('WORKER exit', { id: worker.id, pid: worker.process.pid, code, signal });
             new NoobaaEvent(NoobaaEvent.FORK_EXIT).create_event(undefined, { id: worker.id, pid: worker.process.pid,
                 code: code, signal: signal}, undefined);
+            // SIGABRT is used for abnormal program termination - https://www.qnx.com/developers/docs/6.4.1/dinkum_en/ecpp/stdlib.html
+            // we saw it happen both in case of no memory and on too much processes started
```
Comment (romayalon) [inline src/util/fork_utils.js:45]: what about other signals? I remember we talked about if we see 2 panics in 10 minutes we will also exit the main process

File: src/util/fork_utils.js (hunk 1 above)
Comment (jackyalbo) [reply src/util/fork_utils.js]: Not relevant anymore - we changed the solution to exit on too many fork failures.

File: src/util/fork_utils.js (hunk 2 above)
Comment (jackyalbo) [reply src/util/fork_utils.js]: added the error

File: src/util/fork_utils.js (hunk 3 above)
Comment (jackyalbo) [reply src/util/fork_utils.js]: Not relevant anymore - we changed the solution to exit on too many fork failures.

File: src/util/fork_utils.js [hunk 4]
```diff
@@ -38,11 +40,27 @@ function start_workers(metrics_port, count = 0) {
...
         cluster.on('exit', (worker, code, signal) => {
-            console.warn('WORKER exit', { id: worker.id, pid: worker.process.pid, code, signal }, 'starting a new one.');
+            console.warn('WORKER exit', { id: worker.id, pid: worker.process.pid, code, signal });
             new NoobaaEvent(NoobaaEvent.FORK_EXIT).create_event(undefined, { id: worker.id, pid: worker.process.pid,
                 code: code, signal: signal}, undefined);
+            // This code part will check if we god too many exit events on forks being killed
+            // if we get more than NSFS_MAX_EXIT_EVENTS_PER_TIME_FRAME in a time frame of NSFS_MAX_EXIT_EVENTS_PER_TIME_FRAME
+            // we will kill the main process and stop creating new forks.
+            const now = Date.now();
+            while (exit_events.length && now - exit_events[0] > config.NSFS_EXIT_EVENTS_TIME_FRAME_MIN * 60 * 1000) {
```
Comment (romayalon) [inline src/util/fork_utils.js:50]: please add a comment

**Review by romayalon — APPROVED** [review]

**Pull request merged**